app/jobs/regular/sync_acls_for_uploads.rb

```diff
@@ -18,7 +18,18 @@ def execute(args)
         Rails.logger.warn("Syncing ACL for upload ids: #{args[:upload_ids].join(", ")}")
         Upload.includes(:optimized_images).where(id: args[:upload_ids]).find_in_batches do |uploads|
           uploads.each do |upload|
-            Discourse.store.update_upload_ACL(upload, optimized_images_preloaded: true)
+            begin
+              Discourse.store.update_upload_ACL(upload, optimized_images_preloaded: true)
```

Comment:
Do you know if S3 api allows for bulk update of ACLs?

Reply:
Not easily, but it does look like we can use the S3 batch operations / jobs https://docs.amazonaws.cn/en_us/AmazonS3/latest/userguide/batch-ops.html

For example there is one to replace every ACL in the provided manifest (which would be a list of keys) https://docs.amazonaws.cn/en_us/AmazonS3/latest/userguide/batch-ops-put-object-acl.html

This warrants further exploration I think, I will open a topic internally, but it's out of scope for this PR